content/doc/book/managing/casc.adoc

```diff
@@ -0,0 +1,287 @@
+
+[NOTE]
+====
+Traditionally, experienced Jenkins administrators
+created Apache Groovy `init` scripts
+to automate the configuration for their Jenkins instances.
+This works but requires in-depth understanding of Jenkins APIs
+and the ability to write Groovy scripts.
+Such scripts are powerful and can do almost anything,
+but they also provide few protections against configuration errors.
+====
+
+JCasC provides the convenience and flexibility
+of configuring controllers without using the UI.
+It does not require more understanding of the configuration parameters
+than is required to configure Jenkins through the UI
+and it provides some checks on the values that are provided.
+
+The JCasC configuration file can be checked into an SCM,
+which enables you to determine who made what modifications to the configuration
+and to roll back to a previous configuration if necessary.
+
+The
+link:https://plugins.jenkins.io/configuration-as-code[Configuration as Code] plugin
+must be installed on the Jenkins controller
+that you will use to build out your JCasC configuration.
+If you do not see the *Configuration as Code* tile
+in the *System Configuration* section
+of the *Manage Jenkins* page on your dashboard,
+you need to install the plugin.
+
+== Viewing the JCasC File
+
+When the Configuration as Code plugin is installed,
+you will see *Configuration as Code* in the *System Configuration* section
+of the *Manage Jenkins* page on your dashboard.
+Click on this link,
+then click on *View Configuration*
+to view the YAML file.
+
+This file is an export of the current configuration on this controller.
+In most cases, it is ready to use without modification
+although you usually want to customize it before deploying it.
+You may, however, want to push the unmodified version to SCM
+so you have it as part of your history.
+
+[NOTE]
+====
+The *Configuration as Code* UI page shows the full pathname of the YAML file being used
+and gives a box where you can specify a different file to use.
+See the information below about how to modify the location used for JCasC YAML files.
+====
+
+The default JCasC YAML file has four sections:
+
+* `jenkins` section defines the root Jenkins object,
+with configurations that can be set with the
+*Manage Jenkins >> Configure System*
+and *Manage Jenkins >> Configure Nodes and Clouds* screens.
+
+* `tool` section defines build tools that can be set on the
+*Manage Jenkins >> Global Tool Configuration* screen.
+
+* *unclassified* section defines all other configuration,
+including configuration for installed plugins.
+
+* `credentials` section defines credentials that can be set on the
+*Manage Jenkins >> Manage Credentials* screen.
+You may want to delete this section from your YAML file;
+this is discussed beginning at 27:53 in
+link:https://www.youtube.com/watch?v=ANU7jkxbZSM[How to Install Jenkins on CentOS 7 Using Ansible and JCasC].
+
+== YAML File Syntax
```

Comment:
The [style guide](https://github.com/jenkins-infra/jenkins.io/blob/master/STYLEGUIDE.adoc#assorted-comments) requests sentence case the "titles" (I interpret that to mean "headings", but that is not a consistent usage throughout the documents, for sure) (optional)
```suggestion
== YAML file syntax
```